src/components/right-side/index.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from "react";
+import { NavLink, useLocation } from "react-router-dom";
+
+const RightSide = ({ menu, activeIndex }) => {
+  const location = useLocation();
+  useEffect(() => {
+    const hash = location.hash;
+    console.log(hash.substring(1));
+    const header = document.getElementById(hash.substring(1));
+
+    header && scrollToTarget(header, 70);
+
+    function scrollToTarget(element, offset) {
+      const elementPosition = element.getBoundingClientRect().top;
+      const offsetPosition = elementPosition + window.scrollY - offset;
+
+      window.scrollTo({
+        top: offsetPosition,
+        behavior: "smooth",
+      });
+    }
+  }, [location]);
+  return (
+    <div className="hidden sticky top-20 mt-40 lg:flex flex-col h-fit w-[25%]">
+      <p className="text-left mb-5">From this article</p>
+      <div className="w-full h-full border-l border-gray-500 text-left pl-3">
+        <ul className="space-y-3">
+          {menu.map((item, index) => (
+            <li
+              key={index}
```

Comment:
@chaldrak 
C'est déconseillé d'utiliser les `index` comme key. Si tu le veux, on échange déçu

Reply:
- ça je veux bien un peu plus de détails.
- En fait ce que moi j'avais compris de la clé ça permettait de rendre unique chaque item, et par la même occasion permettre à react de les identifier individuellement. Right ?
- C'est donc pour ça que j'ai l'habitude d'utiliser les index parce que pour eux au moins suis sûr de l'unicité.

Reply:
@chaldrak 
Oui en effet les `index` permettent à React d'identifier les éléments de façons uniques si on lui founi une valeur correcte.
Les index d'un tableau ne sont pas vraiment uniques aux valeurs du tableau. 
Prenons ce tableau 
```js
const students = ["alice","bob","john"];
```
Les index respectifs à ces éléments sont 0 , 1 et 2 nous allons considérer les index comme unique si quelque soit les modifications subit par le tableau alice à toujours l'index 0, bob => 1 et john => 2.
Testons cela : enlevons alice du tableau cela fait, bob prend désormais l'index => 0 et john => 1 tu vois :wink:  les index viennent de changes pour bob et john on peut donc dire que les index ne les permet pas de les identifiés de façons unique.

**Pourquoi cela est un problème pour React ?:**
React pour mettre ajour le UI fait une comparaison de state, right ?
Prenons un exemple sur l'interface, il est dessiné trois personnes :
- Lionel ( un amis a nous deux )
- Moi
- Toi
Supposons qu'on utilise les index pour nous référence, alors l'index pour Lionel est  0 , moi => 1 et toi => 2.
Et bom, il y a changement de state React dois nous redessine, alors what change ? Lionel est retiré de la liste alors je deviens le premier de la liste a l'index 0. Tu es donc a l'index 1 maintenant. React fera la comparaison entre l'element qui a pour cle `0` de son ancien etat a l'element qui a pour cle `0` dans le second etat, c-a-d, il compare Lionel à moi il verra donc qu'il y a eu plusieurs changement la taille, le teint , la coiffure, la corpulence etc... Pauvre React au fait, il va tout simplement transforme Lionel en moi. Il continue son chemain, now il compare l'element a key 1 de l'ancien state a l'element au key 1 du nouveau state plus précisément Moi et toi. Il verra encore qu'il à trop de changement et les appliquera tous, le pauvre. 
Maintenant imaginons qu'aux lieux d'utiliser les index comme key, on utilise nos empreintes respectives. 
Alors que se passerait-il dans React quand il y a changement de state. React prend l'ancien state, son premier element c-a-d Lionel, la key qui lui est associé ( l'empreinte de Lionel ) , React va essayer de retrouver l'element avec le meme empreinte, mais en vain car Lionel est retiré de la liste. React va donc supprimer Lionel de l'UI :wink: . Il passe au deuxième élément c-a-d moi, prend mon empreinte et va essayer dans le retrouve dans le nouveau state et il me retrouvera à la première place. Il fera en suite une comparaison et verra qu'il n'y a pas de changement donc il conserve mon dessin. And so on...

On peut en conséquence conclure qu'utilise les index comme key peut trompe react dans sa comparaison des state et il fera donc des mise-a-jour inutile et couteux ce qui réduire la performance. Et dans les cas extrême peut créer des incohérences dans les rendus,

Reply:
Waooo. J'ai jamais compris ça de cette manière. C'est vraiment ouf.
Aujourd'hui j'ai eu à faire un Map sur les td d'un tableau et j'ai utilisé l'index comme clé.
Rapidement j'ai remarqué vu qu'il y avait plusieurs tr, React a utilisé les même clés deux fois et l'appli a cessé de fonctionner.

Reply:
Si je comprends bien, la clé doit être unique et fixe.
Ça ne doit pas être une valeur qui change d'un chargement à un autre comme un random par exemple

Reply:
@chaldrak 
Oui exactement la clé doit unique à un élément et fixe.
Pour tes `ts` tu peux utiliser les `id` des éléments que tu listes par exemple.

Reply:
Okay.